src/Entity/ContentItem.php

```diff
@@ -35,22 +35,33 @@ public function getShortType($infix = '-')
      *
      * @param ContentItemInterface $from
      * @param ContentItemInterface $to
-     * @return ContentItemInterface
+     * @return array
      */
     public static function convert(ContentItemInterface $from, ContentItemInterface $to)
     {
-        $reflectionFrom = new \ReflectionClass($from);
-        $reflectionTo = new \ReflectionClass($to);
-
-        foreach ($reflectionFrom->getProperties() as $property) {
-            $property->setAccessible(true);
-            $method = 'set' . ucfirst($property->getName());
-            if ($reflectionTo->hasMethod($method)) {
-                $to->$method($property->getValue($from));
-            }
+        $properties = [];
+        try {
+            $reflectionFrom = new \ReflectionClass($from);
+            $reflectionTo = new \ReflectionClass($to);
+
+            do {
+                $convertToType = [];
+                /* @var $p \ReflectionProperty */
+                foreach ($reflectionFrom->getProperties() as $property) {
+                    $property->setAccessible(true);
+                    $method = 'set' . ucfirst($property->getName());
+
+                    if ($reflectionTo->hasMethod($method)) {
+                        $to->$method($property->getValue($from));
+                        $convertToType[$method] = $to->$method($property->getValue($from));
+                    }
+                }
+                $properties = array_merge($convertToType, $properties);
+            } while ($reflectionFrom = $reflectionFrom->getParentClass());
+        } catch (\ReflectionException $e) {
         }
 
-        return $to;
+        return $properties;
```

Comment:
Waarom wil je de data die wordt ge-returned aanpassen? Hiermee verander je ook de signature van een public method. Wat dan strict gezien "breaking" is.

Binnen de page bundle wordt de ge-returnde waarde trouwens niet "afgevangen" of ook maar iets mee gedaan, maar je weet niet of dat ergens anders wel gebeurt.

Reply:
Nee, dit zou nog steeds `$to` moeten retourneren

Reply:
Aangepast